server/src/main/resources/jcasbin_auth_model.conf

```
@@ -0,0 +1,15 @@
+[request_definition]
+r = sub, obj, act
+
+[policy_definition]
+p = sub, obj, act
+
+[role_definition]
+g = _, _
+g2 = _, _
+
+[policy_effect]
+e = some(where (p.eft == allow))
+
+[matchers]
+m = (r.act != 'OWNER' && p.act == r.act && g(p.sub, r.sub) && g2(p.obj, r.obj)) || (r.act == 'OWNER' && p.act == 'OWNER' && r.sub == p.sub && r.obj == p.obj)
```

Shouldn’t the subject comparison be `g(p.sub, r.sub)` in the OWNER clause (same as the non-owner one)?

```suggestion
m = (r.act != 'OWNER' && p.act == r.act && g(p.sub, r.sub) && g2(p.obj, r.obj)) || (r.act == 'OWNER' && p.act == 'OWNER' && g(p.sub, r.sub) && r.obj == p.obj)
```

---

Trying to understand with an example:

When UC has roles, let's say we have a role `R` assigned to a user `U`, with `R` being an `OWNER` of a catalog `C.` A request from `U` should be allowed `OWNER` access to the catalog `C`, correct?

The Casbin DB has these two entries

- `p(R, C, OWNER)`
- `g(R, U)`

And the request has `r(U, C, OWNER)` -- how would this match with the `r.sub == p.sub` check?